### DISCUSSION TITLE
What does it mean when seldepth < depth?

### DISCUSSION THREAD

**Discussion by RogerThiede** [official-stockfish/Stockfish#4360] [Q&A]
Selective Depth, also known as `seldepth`, is the depth of the deepest principal variation line in the search.  Normally, depth and seldepth are either the same or seldepth > depth.  I've been encountering some deep analysis where depth keeps increasing while seldepth actually decreases.
I first was only seeing it when cp = 0, but now I've been seeing it when cp != 0 as shown in multipv 5.
What are the conditions in which the algorithm would cause a decrease of seldepth?

`info depth 74 seldepth 7 multipv 1 score cp 0 wdl 7 986 7 nodes 1089469343535 nps 58897270 hashfull 647 tbhits 2397488 time 18497790 pv ...`
`info depth 74 seldepth 44 multipv 2 score cp 0 wdl 7 986 7 nodes 1089469343535 nps 58897270 hashfull 647 tbhits 2397488 time 18497790 pv ...`
`info depth 74 seldepth 82 multipv 3 score cp 0 wdl 7 986 7 nodes 1089469343535 nps 58897270 hashfull 647 tbhits 2397488 time 18497790 pv ...`
`info depth 74 seldepth 91 multipv 4 score cp -3 wdl 6 986 8 nodes 1089469343535 nps 58897270 hashfull 647 tbhits 2397488 time 18497790 pv ...`
`info depth 74 seldepth 72 multipv 5 score cp -8 wdl 5 985 10 nodes 1089469343535 nps 58897270 hashfull 647 tbhits 2397488 time 18497790 pv ...`
`info depth 74 seldepth 79 multipv 6 score cp -13 wdl 3 983 14 nodes 1089469343535 nps 58897270 hashfull 647 tbhits 2397488 time 18497790 pv ...`

**Comment by vondele** [comment]
It means at a certain depth it is not able to find a continuation that stays within the alpha-beta bounds, i.e. fails high or low, and as such is not on the PV.
